File: app/src/main/java/com/example/myanimedata/adapter/RoleAdapter.java
package com.example.myanimedata.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myanimedata.R;
import com.example.myanimedata.api.RoleResult;

import java.util.List;

public class RoleAdapter extends RecyclerView.Adapter<RoleAdapter.RoleViewHolder>{
    private final List<RoleResult> roleResults;

    public RoleAdapter(List<RoleResult> roleResults) {
        this.roleResults = roleResults;
    }

    @NonNull
    @Override
    public RoleAdapter.RoleViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new RoleAdapter.RoleViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.container_genre, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RoleAdapter.RoleViewHolder holder, int position) {
        holder.bindItem(roleResults.get(position));
    }

    @Override
    public int getItemCount() {
        return roleResults.size();
    }

    public static class RoleViewHolder extends RecyclerView.ViewHolder{
        private final TextView textRole;

        public RoleViewHolder(@NonNull View itemView) {
            super(itemView);
            this.textRole = itemView.findViewById(R.id.textGenres);
        }

        public void bindItem(RoleResult roleResult) {
            textRole.setText(roleResult.getAnimeRoleResult().getTitle());
        }
    }
}

File: app/src/main/java/com/example/myanimedata/adapter/RoleAdapter.kt
package com.example.myanimedata.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.myanimedata.R
import com.example.myanimedata.adapter.RoleAdapter.RoleViewHolder
import com.example.myanimedata.api.RoleResult
import com.google.android.material.chip.Chip

class RoleAdapter(private val roleResults: List<RoleResult>) :
    RecyclerView.Adapter<RoleViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RoleViewHolder {
        return RoleViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.container_genre, parent, false)
        )
    }

    override fun onBindViewHolder(holder: RoleViewHolder, position: Int) {
        holder.bindItem(roleResults[position])
    }

    override fun getItemCount(): Int {
        return roleResults.size
    }

    class RoleViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val textRole: Chip

        init {
            textRole = itemView.findViewById(R.id.textGenres)
        }

        fun bindItem(roleResult: RoleResult) {
            textRole.text = roleResult.animeRoleResult.title
        }
    }
}